wordpress-mobile/wellsql
# Upgrade Gradle to 7.1.1 & Android Gradle Plugin to 4.2.2

This PR prepares the repository for publishing its artifacts to our S3 Maven. The main goal of the PR is to upgrade Gradle to `7.1.1` and Android Gradle Plugin to `4.2.2`, the rest of the changes are side effects. Here is the list of changes:

* Updates Gradle to `7.1.1` & Android Gradle Plugin to `4.2.2`
* Removes deprecated bintray & android-maven plugins. These will be replaced with our publishing plugin.
* Removes the unnecessary `buildToolsVersion` declaration. This is automatically assigned with the new versions of AGP.
* Updates `compileSdkVersion` & `targetSdkVersion` to `29` as this version of AGP does not work with version `28`.
* Removes deprecated `com.android.support` dependency from the sample app, as version `29` for it doesn't exist, and replaces it with the new `androidx.appcompat` one.
* Removes bintray properties from `gradle.properties.example` and adds `android.useAndroidX=true` instead. The new key is required due to `androidx.appcompat` dependency.
* Removes bintray publishing section from `README`. Further updates to `README` will be made in the upcoming PR for publishing to S3.

**To test:**
* Running `./gradlew build` should be successful.

---

I am marking this PR as draft so it can be merged in sync with the publish to s3 PR since this PR will break Jitpack builds. It'd be great if it can be reviewed in advance.
